src/download.rs

```diff
@@ -214,84 +191,11 @@ impl Session {
     }
 }
 
-impl MimeMessage {
-    /// Creates a placeholder part and add that to `parts`.
-    ///
-    /// To create the placeholder, only the outermost header can be used,
-    /// the mime-structure itself is not available.
-    ///
-    /// The placeholder part currently contains a text with size and availability of the message.
-    pub(crate) async fn create_stub_from_partial_download(
-        &mut self,
-        context: &Context,
-        org_bytes: u32,
-    ) -> Result<()> {
-        let mut text = format!(
-            "[{}]",
-            stock_str::partial_download_msg_body(context, org_bytes).await
-        );
-
-        info!(context, "Partial download: {}", text);
-
-        self.do_add_single_part(Part {
-            typ: Viewtype::Text,
-            msg: text,
-            ..Default::default()
-        });
-
-        Ok(())
-    }
-}
-
 #[cfg(test)]
 mod tests {
-    use num_traits::FromPrimitive;
-
     use super::*;
-    use crate::chat::{get_chat_msgs, send_msg};
-    use crate::ephemeral::Timer;
-    use crate::message::delete_msgs;
-    use crate::receive_imf::receive_imf_from_inbox;
-    use crate::test_utils::{E2EE_INFO_MSGS, TestContext, TestContextManager};
-
-    #[test]
-    fn test_downloadstate_values() {
-        // values may be written to disk and must not change
-        assert_eq!(DownloadState::Done, DownloadState::default());
-        assert_eq!(DownloadState::Done, DownloadState::from_i32(0).unwrap());
-        assert_eq!(
-            DownloadState::Available,
-            DownloadState::from_i32(10).unwrap()
-        );
-        assert_eq!(DownloadState::Failure, DownloadState::from_i32(20).unwrap());
-        assert_eq!(
-            DownloadState::InProgress,
-            DownloadState::from_i32(1000).unwrap()
-        );
-    }
-
-    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
-    async fn test_download_limit() -> Result<()> {
```

Comment:
The same, maybe leave it for now?

Reply:
Then the `Config` key should be removed or at least deprecated

Reply:
Because this download_limit is only for deciding whether to download full-messages, it does not affect any other messages anymore.

Those other messages will get downloaded always automatically in foreground, and in background only if they are smaller than one megabyte.